lib/pact_broker/pacts/service.rb

```diff
@@ -95,6 +95,10 @@ def find_latest_pacts_for_provider provider_name, options = {}
         pact_repository.find_latest_pacts_for_provider provider_name, options[:tag]
       end
 
+      def find_latest_pacts_for_provider_for_branch provider_name, options = {}
+        pact_repository.find_latest_pacts_for_provider provider_name, options[:branch]
```

Comment:
I think this uses 

https://github.com/pact-foundation/pact_broker/blob/87669c36351df60b4dfecaf8996073fb0382fadc/lib/pact_broker/pacts/repository.rb#L146C25-L146C48

https://github.com/pact-foundation/pact_broker/blob/master/lib/pact_broker/pacts/pact_publication_dataset_module.rb#L235

I am not super well versed with the queries so correct me if I am wrong, but I think this is old way of querying?

https://docs.pact.io/blog/2021/07/04/why-we-are-getting-rid-of-tags

instead should we try to use logic based on the below method ?

https://github.com/pact-foundation/pact_broker/blob/master/lib/pact_broker/pacts/pact_publication_dataset_module.rb#L144